src/java/org/apache/cassandra/index/sai/view/RangeTermTree.java

```diff
@@ -86,10 +86,13 @@ public void addIndex(SSTableIndex index)
 
             if (logger.isTraceEnabled())
             {
+                if (indexTermType.isVector())
+                    throw new IllegalArgumentException("This is a bug, please, report.");
+                
```

Comment:
On a second thought this will be changed to: `assert indexTermType.isVector();`

Reply:
Yep, this should probably be at the beginning of the method. Also, `addIndex` could be inlined inside `add`, which is its only caller.